### DISCUSSION TITLE
Native Support for PWAs

### DISCUSSION THREAD

**Discussion by DhananjayPorwal** [zen-browser/desktop#2690] [Polls]
Currently, Chromium browsers support Progressive Web Apps (PWAs) natively, but Firefox doesn’t offer the same experience.

Should Zen Browser add native support for PWAs?

Key features could include:

+ Easy installation of PWAs from the browser
+ Add PWAs to the home screen or desktop
+ Push notifications and offline support
+ Simple management of installed PWAs

> I know that firefox extension "Progressive Web Apps for Firefox" has ability to create PWA

**Comment by Smooth-E** [comment]
1 up for this, as [PWAs for Firefox](https://github.com/filips123/PWAsForFirefox) works by using a separate Firefox runtime which updates independently. It also brings a separate profile with separate set of extensions and settings, which you can synchronize with Firefox Accounts, but it is still not that convenient.

**Reply by DhananjayPorwal** [reply]
If Zen could create one using Zen's UI, it would be incredibly useful.

**Reply by erkkimon** [reply]
+1

Make PWAs great again! Many PWAs and SPAs get often require opening links in default browser to be usable, which often breaks those apps. Also opening urls assigned to a PWA should open in the PWA, which never seems to work when using a plugin.

PWA and SPA are also technologies that have made it very possible to use GNU/Linux at work, but there are very many rough edges with that plugin. 

Zen Browser definitely could make a difference!

**Reply by andre-brandao** [reply]
It would also be great to have a cli flag similar to --app=<website_url> which exists in Chromium browsers, making it easier to integrate with special workspaces and scratchpads in Hyprland and other similar workflows

**Comment by torgeros** [comment]
Mozilla is working on this upstream: https://firefox-source-docs.mozilla.org/browser/components/taskbartabs/docs/index.html

This already works in the latest Zen, but is very unintuitive because it opens in a new window with pinned tabs there but the window layout (e.g.) being different.

Either way, it is probably not worth to implement a Zen-specific solution right now ^^

**Reply by Pshemas** [reply]
well, seeing how Mozilla tackles this I'd say I would be very happy to see a zen touch for this feature. So maybe waiting for Moz to implement it fully then giving it a chef's kiss would be nice ;) .

**Comment by markcst** [comment]
Hi! No updates on this?

**Comment by mr-cheffy** [comment]
First step is getting little zen done: https://github.com/zen-browser/desktop/pull/13450

and being able to use that as a gateway for PWAs after

**Reply by markcst** [reply]
Hi! Great!
Could you please describe what's the idea behind little zen? What is that?

**Reply by AfeefurR** [reply]
its similar to little arc. basically being able to spawn a small window (usually temporary, but you can move it into your main spaces if you wish) from anywhere in the system.
